AlphaWallet/Tokens/Coordinators/GetENSOwnerCoordinator.swift

```diff
@@ -0,0 +1,90 @@
+//
+// Created by James Sangalli on 8/11/18.
+//
+import Foundation
+import Result
+import web3swift
+import CryptoSwift
+
+//https://github.com/ethereum/EIPs/blob/master/EIPS/eip-137.md
+extension String {
+    public var nameHash: String {
+        var node = Array<UInt8>.init(repeating: 0x0, count: 32)
+        if !self.isEmpty {
+            node = self.split(separator: ".")
+                .map { Array($0.utf8).sha3(.keccak256) }
+                .reversed()
+                .reduce(node) { return ($0 + $1).sha3(.keccak256) }
+        }
+        return "0x" + node.toHexString()
+    }
+}
+
+class GetENSOwnerCoordinator {
+
+    private let config: Config
+
+    init(config: Config) {
+        self.config = config
+    }
+
+    func getENSOwner(
+            for input: String,
+            completion: @escaping (Result<EthereumAddress, AnyError>) -> Void
+    ) {
+
+        //if already an address, send back the address
+        if let ethAddress = EthereumAddress(input) {
+            completion(.success(ethAddress))
+            return
+        }
+
+        //if it does not contain .eth, then it is not a valid ens name
+        if !input.contains(".eth") {
```

Comment:
We can use `hasSuffix(".eth")` instead. But it appears that there's also .xyz and .luxe. Maybe we can't perform this check reliably?